src/hotspot/share/opto/gcm.cpp

```diff
@@ -729,29 +772,44 @@ Block* PhaseCFG::insert_anti_dependences(Block* LCA, Node* load, bool verify) {
     }
   }
 
-  ResourceArea* area = Thread::current()->resource_area();
-  DefUseMemStatesQueue worklist_def_use_mem_states(area); // prior memory state to store and possible-def to explore
-  Node_List non_early_stores(area); // all relevant stores outside of early
-  bool must_raise_LCA = false;
+  assert(early->dominates(LCA_orig), "precondition failed");
 
-  // 'load' uses some memory state; look for users of the same state.
-  // Recurse through MergeMem nodes to the stores that use them.
+  ResourceArea* area = Thread::current()->resource_area();
 
-  // Each of these stores is a possible definition of memory
-  // that 'load' needs to use.  We need to force 'load'
-  // to occur before each such store.  When the store is in
-  // the same block as 'load', we insert an anti-dependence
-  // edge load->store.
+  // Bookkeeping of possibly anti-dependent stores that we find outside of the
+  // early block and that may need anti-dependence edges. For efficiency, we
+  // use a lazy approach to add anti-dependence edges, and only add them at the
+  // very end when we know the final updated LCA.
+  Node_List non_early_stores(area);
 
-  // The relevant stores "nearby" the load consist of a tree rooted
-  // at initial_mem, with internal nodes of type MergeMem.
-  // Therefore, the branches visited by the worklist are of this form:
-  //    initial_mem -> (MergeMem ->)* Memory state modifying node
-  // Memory state modifying nodes include Store and Phi nodes and any node for which needs_anti_dependence_check()
-  // returns false.
-  // The anti-dependence constraints apply only to the fringe of this tree.
+  // Flag that indicates if we must attempt to raise the LCA after the main
```

Comment:
```suggestion
  // Whether we must attempt to raise the LCA after the main
```
Also, could you clarify what you mean by "attempt"? Could LCA raising fail somehow?